### ISSUE TITLE
Add test coverage for update events in e2e tests

### ISSUE THREAD

**Issue by PrasadG193** [kubeshop/botkube#341]
In e2e test suite, we mock resource creation with fake client and verify the BotKube event generations.
https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go#L46

We need to add similar test cases to verify update event handling.
We can use [Patch](https://godoc.org/k8s.io/client-go/kubernetes/typed/core/v1#PodInterface) API for generating the update event. Along with update notification verification, we should also test if diff is getting calculated and passed with notification correctly

Resource configuration can be mocked by setting/altering values in `utils.AllowedUpdateEventsMap` and `utils.AllowedEventKindsMap` https://github.com/infracloudio/botkube/blob/develop/pkg/utils/utils.go#L148

Add test cases to validate the following behavior:

- [x] Validate update events are handled for a resource with `events: [all]` and `updateSettings` in test config https://github.com/infracloudio/botkube/blob/develop/test/resource_config.yaml. - @A-kanksh-a 

- [x] Validate update events are handled for a resource with `events: [all]` without `updateSettings` in test config https://github.com/infracloudio/botkube/blob/develop/test/resource_config.yaml. - @A-kanksh-a 

- [ ] Validate error events are handled with only `events: [update]` config for a resource

- [ ]  Validate error events are skipped when the `update` event is not configured for a resource in test config. e.g `events: [create, error, delete]`. 

- [x] Validate that update events are skipped when an event occurs for a namespace which is not configured - @A-kanksh-a 

- [x] Validate that update events are skipped when an event occurs for a resource which is not added in resource_config - @A-kanksh-a 

- [ ] Validate that update events are skipped when invalid fields are passed in updateSetting (e.g `status.invalid` for Pod).

Each test case can be added as a separate PR. Multiple PR as expected for this task. 
Guidelines for contributors:
- Post comment on the issue informing others about the test case you want to add
- Mark the issue as done once the PR is merged
- Please make sure that no one else is working on the same test case to avoid duplicate PRs

**Comment by PrasadG193** [comment]
@A-kanksh-a I have updated the description with required test cases. Please add the comments about tests cases you will be covering

**Comment by A-kanksh-a** [comment]
@PrasadG193 I am working on the following:-

- Validate update events are handled for a resource with `events: [all]` and `updateSettings` in test config - 
- Validate update events are handled for a resource with `events: [all]` without `updateSettings` in test config
- Validate that update events are skipped when an event occurs for a namespace which is not configured
- Validate that update events are skipped when an event occurs for a resource which is not added in resource_config

I have raised a PR #351 for first 2 test scenarios.

**Comment by A-kanksh-a** [comment]
@PrasadG193 I will be contributing to one more testcase:-

- Validate that update events are skipped when invalid fields are passed in updateSetting (e.g status.invalid for Pod).

**Comment by avats-dev** [comment]
@PrasadG193 I want to work on :
- Validate error events are skipped when the update event is not configured for a resource in test config. e.g `events: [create, error, delete]`

Also, can you guide me a bit in implementing this as I'm not that well-versed in `go`?

Maybe @A-kanksh-a can as she has resolved many of these.

**Comment by A-kanksh-a** [comment]
@avats-dev you can look into the following file for reference:- [https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/update/update.go](https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/update/update.go) or [https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go](https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go)

**Comment by pkosiec** [comment]
Hi, 
I'm closing this task as not valid anymore, as we switched to the E2E tests with real Slack and Kubernetes APIs and we have covered the feature in E2E and unit tests.
We can discuss new ideas under this epic: https://github.com/kubeshop/botkube/issues/589.

Thanks everyone involved!
